src/sha1_hash.cpp

```diff
@@ -65,41 +66,11 @@ namespace libtorrent
 
 	int sha1_hash::count_leading_zeroes() const
 	{
-		int ret = 0;
-		for (int i = 0; i < number_size; ++i)
-		{
-			std::uint32_t v = aux::network_to_host(m_number[i]);
-			if (v == 0)
-			{
-				ret += 32;
-				continue;
-			}
-
-#if TORRENT_HAS_BUILTIN_CLZ
-			return ret + __builtin_clz(v);
-#elif TORRENT_HAS_SSE && defined _MSC_VER
-			DWORD pos;
-			_BitScanReverse(&pos, v);
-			return ret + 31 - pos;
+#if TORRENT_HAS_BUILTIN_CLZ || defined _MSC_VER
```

Comment:
here. everywhere clz_* is called, I imagine these macros would have to be checked. and if another implementation is added, all these places would have to be updated. another function that picks the right implementation might make sense